lib/common/string.c

```diff
@@ -421,11 +421,11 @@ const char *h2o_next_token(h2o_iovec_t *iter, int separator, size_t *element_len
     for (;; ++cur) {
         if (cur == end)
             break;
-        if (*cur == separator) {
+        if (*cur == inner) {
             ++cur;
             break;
         }
-        if (*cur == ',') {
+        if (outer && *cur == outer) {
```

Comment:
Do we need to check outer?

The question is what the caller should set, when parsing a non-nested structure. I actually see two ways being used in this PR. One sets `outer` to `0`. The other sets the same character as `inner`.

I think we should clarify which approach is appropriate (in the doc-comment of string_.h), and use that approach throughout our code-base.

Regarding which approach is preferable, my not-so-strong preference goes to the latter, as it removes this special case, and because it's more consistent with our approach with allowing any character, including NUL, to be used in a string.